docs/developers_guide/FlinkComponents.md

```diff
@@ -1,22 +1,72 @@
 
 :warning: **Flink components should not extend Lifecycle** - it won't be handled properly
 
 ## Sources
 
-Source are defined with [FlinkSource](https://github.com/TouK/nussknacker/blob/staging/engine/flink/components-api/src/main/scala/pl/touk/nussknacker/engine/flink/api/process/FlinkSource.scala).
-In most cases (when you only pass one variable to initial `Context`) it's easier to use `BaseFlinkSource`.
+Sources implementations are defined with [FlinkSource](https://github.com/TouK/nussknacker/blob/staging/engine/flink/components-api/src/main/scala/pl/touk/nussknacker/engine/flink/api/process/FlinkSource.scala).
+In most cases (when you only pass one variable to initial `Context`) it's easier to use `BasicFlinkSource`. This trait
+requires only Flink's `SourceFunction` to be implemented, whereas `FlinkSource` allows to prepare an arbitrary `DataStream`
+and add custom [context initializer](https://github.com/TouK/nussknacker/blob/staging/components-api/src/main/scala/pl/touk/nussknacker/engine/api/process/ContextInitializer.scala).
+You should bear in mind to apply a watermark strategy to the stream so that events will be processed correctly downstream, e.g. in aggregates.
+
+Sources usually emit a single variable named `#input` having some attributes. To have those attributes recognized
+by the designer you need to specify them. For more information consult [components specification](Components.md#specification)
+and [types](Basics.md#types) to describe (available fields, their types) the emitted `#input`.
+
+### Test data generation
+
+Sources optionally support generating test data to ease [scenarios development](../scenarios_authoring/TestingAndDebugging.md).
+In order to support this feature, `FlinkSourceTestSupport` trait has to be implemented by the source.
+
+### Integrating with Flink sources
+
+Flink comes with simple sources [already predefined](https://ci.apache.org/projects/flink/flink-docs-master/docs/dev/datastream/overview/#data-sources)
+and [connectors](https://ci.apache.org/projects/flink/flink-docs-master/docs/connectors/datastream/overview) with third-party systems.
+All of them can be used to implement a Nussknacker source. Nussknacker has built-in support for reading Kafka records in different formats:
+Avro or Json with schemas defined in Schema Registry - see [`FlinkKafkaComponentProvider`](https://github.com/TouK/nussknacker/blob/staging/engine/flink/components/kafka/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/FlinkKafkaComponentProvider.scala).
+If you do not have Schema Registry in your organization, a source reading records with fixed schema is easily doable,
+for a example `real-kafka-json-SampleProduct` in [`DevProcessConfigCreator`](https://github.com/TouK/nussknacker/blob/staging/engine/flink/management/dev-model/src/main/scala/pl/touk/nussknacker/engine/management/sample/DevProcessConfigCreator.scala)
```

Comment:
I wonder if it's good idea to link to DevConfigCreator or existing Kafka components as a guide. DevConfigCreator contains a lot of strange code that's needed for testing, while existing Kafka components are really complex.
Maybe it would be better to add sth to https://github.com/TouK/nussknacker-sample-components/? We don't have any sample sources/sinks ATM